modules/services/redshift-gammastep/lib/options.nix

```diff
@@ -113,26 +121,64 @@ in {
       '';
     };
 
-    extraOptions = mkOption {
-      type = types.listOf types.str;
-      default = [ ];
-      example = [ "-v" "-m randr" ];
+    settings = mkOption {
+      type = types.submodule { freeformType = settingsFormat.type; };
+      default = { };
+      example = literalExample ''
+        {
+          ${mainSection} = {
+            adjustment-method = "randr";
+          };
+          randr = {
+            screen = 0;
+          };
+        };
+      '';
       description = ''
-        Additional command-line arguments to pass to
-        <command>redshift</command>.
+        The configuration to pass to ${programName}.
+        Available options for ${programName} described in
+        <citerefentry>
+          <refentrytitle>${moduleName}</refentrytitle>
+          <manvolnum>1</manvolnum>
+        </citerefentry>.
       '';
     };
   };
 
   config = {
     assertions = [{
-      assertion = cfg.provider == "manual" -> cfg.latitude != null
-        && cfg.longitude != null;
-      message = "Must provide services.${moduleName}.latitude and"
-        + " services.${moduleName}.latitude when"
-        + " services.${moduleName}.provider is set to \"manual\".";
+      assertion = (cfg.settings ? ${moduleName}.dawn-time || cfg.settings
+        ? ${moduleName}.dusk-time)
+        || (cfg.settings.${moduleName}.location-provider) == "geoclue2"
+        || ((cfg.settings.${moduleName}.location-provider) == "manual"
+          && (cfg.settings ? ${moduleName}.latitude || cfg.settings
+            ? ${moduleName}.longitude));
```

Comment:
Will need to update these to match the new `manual.lat`, `manual.lon` naming.

I'm updating the test suite to catch these types or errors.

Reply:
Good catch. Fixing it.

Reply:
Done.

Reply:
Seems this change already got another issue in this module, this change is definitely welcome :heart: .